models/articles-model.js

```diff
@@ -0,0 +1,21 @@
+const db = require('../db/connection');
+
+exports.fetchArticleByID = (article_id) => {
+    return db
+    .query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+    .then(({rows: articleData}) => {
+        if(articleData.length === 0) {
+            throw new Error;
+        }
+        else {
+            return articleData;
+        }
+    })
+    .catch((err) => {
+       if(err.code === '22P02') {
+        return Promise.reject({status: 400, msg: 'Invalid Data Type'})
+       } 
+       
+       return Promise.reject({status: 404, msg: 'That article does not exist'})
+    })
```

Comment:
Replace throw new Error with your promise.reject from line 19 and then remove the catch block. You don't need a catch block on a model, psql will return that error code to your controller in which case you can catch the error in your controller and pass it to next. And then you can have some error handling middleware that deals with that error code. 

Otherwise you will have to repeat this on every model you do!